build-tools/config/webpack/webpack.partial.conf.plugins.js

```diff
@@ -0,0 +1,152 @@
+            template: 'index.html',
+            inject: true,
+            version: config.build.versionPath,
+            minify: {
+              removeComments: true,
+              collapseWhitespace: true,
+              removeAttributeQuotes: false,
+            },
+            chunksSortMode: 'dependency',
+          }
+        : {
+            filename: 'index.html',
+            template: 'index.html',
+            version: '/',
+            inject: true,
+            cache: true,
+          },
+    ),
+  ];
+
+  if (isDevelopment) {
+    /*
+     * ------------------------------------------------
+     * Development-only plugins
+     * ------------------------------------------------
+     */
+    plugins.push(
+      new webpack.HotModuleReplacementPlugin(),
+      new FriendlyErrorsWebpackPlugin({
+        compilationSuccessInfo: {
+          messages: [
+            `Your application is running here: ${
+              config.useHttps ? 'https' : 'http'
+            }://localhost:${process.env.PORT || config.dev.port}`,
+          ],
+        },
+      }),
+    );
+  } else {
+    /*
+     * ------------------------------------------------
+     * Production-only plugins
+     * ------------------------------------------------
+     */
+    plugins.push(
+      new WebpackCleanupPlugin(),
+      new MiniCssExtractPlugin({
+        filename: path.posix.join(config.build.versionPath, 'css/[name].css'),
+      }),
+      new LodashModuleReplacementPlugin({
+        paths: true,
+      }),
+      new webpack.NamedChunksPlugin(),
+      new OptimizeCssAssetsPlugin({
+        cssProcessorOptions: {
+          parser: require('postcss-safe-parser'),
+          discardComments: {
+            removeAll: true,
+          },
+        },
+      }),
+      new ImageminPlugin({
+        disable: !config.build.enableImageOptimization,
+        svgo: null,
+        gifsicle: null,
+        pngquant: config.build.enablePNGQuant ? { quality: config.build.pngQuantQuality } : null,
+      }),
+    );
+
+    if (config.build.analyze) {
+      plugins.push(
+        new BundleAnalyzerPlugin({
+          defaultSizes: 'gzip',
+        }),
+      );
+    }
+
+    if (config.build.generateIcons) {
+      plugins.push(
+        new FaviconsWebpackPlugin({
+          logo: path.join(config.projectRoot, 'static/image/favicon.png'),
+          prefix: config.build.versionPath + 'static/favicon/',
+          emitStats: false,
+          persistentCache: false,
+          inject: true,
+          background: '#fff',
+          title: '',
+          icons: {
+            android: true,
+            appleIcon: true,
+            appleStartup: false,
+            coast: false,
+            favicons: true,
+            firefox: true,
+            opengraph: true,
+            twitter: true,
+            yandex: true,
+            windows: true,
+          },
+        }),
+      );
+    }
+  }
+
+  return {
+    ...webpackConfig,
+    plugins,
```

Comment:
A suggestion when reading there are few conditional statements. Maybe split development and production configs in a separate defined array.

```
return {
    ...webpackConfig,
   plugins.concat(isDevelopment ? developmentPlugins : productionPlugins)
}
```

We can do this for this case as plugins afaik aren't dependent on order.

Reply:
The difference is that the condition doesn't happen inline. I suggested this to improve readability.